src/main/kotlin/persistence/AppDatabaseImpl.kt

```diff
@@ -0,0 +1,58 @@
+package persistence
+
+import data.Language
+import data.User
+import data.dao.Dao
+import data.persistence.AppDatabase
+import io.requery.Persistable
+import io.requery.sql.KotlinConfiguration
+import io.requery.sql.KotlinEntityDataStore
+import io.requery.sql.SchemaModifier
+import io.requery.sql.TableCreationMode
+import org.sqlite.SQLiteDataSource
+import persistence.model.Models
+import persistence.repo.LanguageRepo
+import persistence.repo.UserLanguageRepo
+import persistence.repo.UserRepo
+import javax.inject.Inject
+
+class AppDatabaseImpl: AppDatabase {
+
+    private val dataStore: KotlinEntityDataStore<Persistable> = createDataStore()
+    private var languageDao: Dao<Language> = LanguageRepo(dataStore)
+    private var userLanguageRepo: UserLanguageRepo = UserLanguageRepo(dataStore)
+    private var userDao: Dao<User> = UserRepo(dataStore, userLanguageRepo, languageDao)
+
+
+    private fun createDataStore() : KotlinEntityDataStore<Persistable> {
+        Class.forName("org.sqlite.JDBC")
+
+        val sqLiteDataSource = SQLiteDataSource()
+        sqLiteDataSource.url = "jdbc:sqlite:content.db"
+
+        // creates tables that do not already exist
+        SchemaModifier(sqLiteDataSource, Models.DEFAULT).createTables(TableCreationMode.CREATE_NOT_EXISTS)
+
+        // sets up data store
+        val config = KotlinConfiguration(dataSource = sqLiteDataSource, model = Models.DEFAULT)
+        return KotlinEntityDataStore<Persistable>(config)
+    }
+
+    override fun getUserDao(): Dao<User> {
+        return userDao
+    }
+
+    override fun getLanguageDao(): Dao<Language> {
+        return languageDao
+    }
+
+    companion object {
+        // initializing here to avoid needing null handling
+        private var instance : AppDatabase = AppDatabaseImpl()
```

Comment:
why not val?

Reply:
I'll change that. I think it was left over from when I was doing `private var instance : AppDatabase? = null`